src/pages/settings/ClusteredSettingFormInput.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC, useState } from "react";
+import { Button, Form, Icon } from "@canonical/react-components";
+import type { ConfigField } from "types/config";
+import { getConfigId } from "./SettingForm";
+import ConfigFieldDescription from "pages/settings/ConfigFieldDescription";
+import ClusterSpecificInput from "components/forms/ClusterSpecificInput";
+import { useClusterMembers } from "context/useClusterMembers";
+import { ClusterSpecificValues } from "components/ClusterSpecificSelect";
+
+interface Props {
+  initialValue: ClusterSpecificValues;
+  configField: ConfigField;
+  onSubmit: (newValue: ClusterSpecificValues) => void;
+  onCancel: () => void;
+  readonly?: boolean;
+  toggleReadOnly?: () => void;
+}
+
+const ClusteredSettingFormInput: FC<Props> = ({
+  initialValue,
+  configField,
+  onSubmit,
+  onCancel,
+  readonly = false,
+  toggleReadOnly,
+}) => {
+  const [value, setValue] = useState<ClusterSpecificValues>(initialValue);
+
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const memberNames = clusterMembers.map((member) => member.server_name);
+
+  const canBeReset = Object.values(value).some(
+    (v) => v !== configField.default,
+  );
+
+  const resetToDefault = (memberNames: string[]) => {
+    const defaultValues: { [key: string]: string } = {};
+    memberNames.forEach((name) => {
+      defaultValues[name] = configField.default;
+    });
+    setValue(defaultValues);
+  };
```

Comment:
```suggestion
  const resetToDefault = () => {
    const defaultValues: { [key: string]: string } = {};
    memberNames.forEach((name) => {
      defaultValues[name] = configField.default;
    });
    setValue(defaultValues);
  };
```
I think we can simplify this a bit.